### ISSUE TITLE
Sound card test failure (head)

### ISSUE THREAD

**Issue by spyhunter99** [oshi/oshi#860]
````
INFO] Running oshi.hardware.SoundCardTest
[ERROR] Tests run: 1, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 0.076 s <<< FAILURE! - in oshi.hardware.SoundCardTest
[ERROR] testSoundCards  Time elapsed: 0.076 s  <<< FAILURE!
java.lang.AssertionError
	at oshi.hardware.SoundCardTest.testSoundCards(SoundCardTest.java:46)

`````

**Comment by spyhunter99** [comment]
this is on centos7 (latest) in a virtual box vm

**Comment by dbwiddis** [comment]
Looks like this line of code is somehow setting the string to null. Can you investigate why this happens?
```
        String cardCodec = "";
        File[] cardFiles = cardDir.listFiles();
        if (cardFiles != null) {
            for (File file : cardFiles) {
                if (file.getName().startsWith("codec")) {
                    cardCodec = FileUtil.getKeyValueMapFromFile(file.getPath(), ":").get("Codec");
                }
            }
        }
```
Presumably there is a file `codec*` that doesn't include "Codec" in the list of keys with values.  What does it include?

**Comment by spyhunter99** [comment]
![image](https://user-images.githubusercontent.com/1975321/57960161-4c00d180-78d5-11e9-8160-d36415d9f6d4.png)

looks like we are trying to read from a directory

**Comment by spyhunter99** [comment]
contents of the directory
![image](https://user-images.githubusercontent.com/1975321/57960186-6b97fa00-78d5-11e9-9b4d-777019063366.png)

**Comment by spyhunter99** [comment]
so if  you are looking for the codec, i think ac97 is what you are looking for. I poked around in `ac97#0-0` and it is a key/value pair file, but there is no `codec` key of any case in it.

**Comment by dbwiddis** [comment]
Fixed in #860
